htmd/parameterization/fftype.py

```diff
@@ -161,13 +182,155 @@ def fftype(mol, rtfFile=None, prmFile=None, method=FFTypeMethod.CGenFF_2b6, acCh
+        self.assertListAlmostEqual(list(self.testMolecule.charge), charges)
+        if len(impropers) != 0:
+            for test, ref in zip(list(self.testMolecule.impropers), impropers):
+                self.assertEqual(list(test), ref)
+        else:
+            self.assertEqual(list(self.testMolecule.impropers), impropers)
+
+    def _test_params(self, params):
+        for i in self.testParameters.__dict__:
+            self.assertEqual(getattr(self.testParameters, i), getattr(params, i))
+
+    def _test_intermediary_files(self, files):
+        self.assertEqual(self.testIntermediaryFiles, files)
+
+    def test_mol(self):
+
+        toTest = (
+            ('ethanolamine', 'GAFF2', 'None'),
+            ('1o5b_ligand', 'GAFF2', ('gas', 1)),
+            ('1z95_ligand', 'GAFF2', ('gas', 0)),
+            ('4gr0_ligand', 'GAFF2', ('gas', -1))
+        )
+
+        for name, method, chargetuple in toTest:
+            with self.subTest(name=name, method=method, chargetuple=chargetuple):
+                refDir = os.path.join(self.refDir, name, method)
+                self._init_mol(name, method, chargetuple)
+                # self._generate_references(name, method)
+                with open(os.path.join(refDir, 'mol_props.yaml'), 'r') as infile:
+                    refProps = self.yamlload(infile)
+                self._test_mol_props(**refProps)
+                with open(os.path.join(refDir, 'params.p'), 'rb') as infile:
+                    refParams = self.pickleload(infile)
+                self._test_params(refParams)
+                with open(os.path.join(refDir, 'intermediary_files.yaml'), 'r') as infile:
+                    refFiles = self.yamlload(infile)
+                self._test_intermediary_files(refFiles)
+
+    def test_broken_atomname(self):
+        molFile = os.path.join(self.refDir, '{}.mol2'.format('ethanolamine_wrongnames'))
+
+        testMolecule = self.Molecule(molFile)
+        testMolecule = self.canonicalizeAtomNames(testMolecule, 'GAFF2')
+        testMolecule, _, _ = self.getEquivalentsAndDihedrals(testMolecule)
+
+        from tempfile import mkdtemp
+        tmpDir = mkdtemp(suffix='fftype')
+        with self.assertRaises(RuntimeError):
+            _, _ = fftype(testMolecule,
+                          method='GAFF2',
+                          acCharges=None,
+                          netcharge=None,
+                          tmpDir=tmpDir)
+
+
 if __name__ == '__main__':
 
+    import sys
+
+    unittest.main(verbosity=2)
+    sys.exit(0)
+
     import sys
     import re
     from htmd.home import home
     from htmd.molecule.molecule import Molecule
-    from htmd.parameterization.writers import writeRTF, writePRM, writeFRCMOD
+    from htmd.parameterization.writers import writeRTF, writePRM
     from htmd.parameterization.util import canonicalizeAtomNames, getEquivalentsAndDihedrals
 
     # BUG: MATCH does not work on Mac!
```

Comment:
I do not want to mess with CGenFF. I am not paying attention to it right now, as it will take more time (which we don't have) to make it work, test, etc. Furthermore, we already know what a mess is the match atom typer